### PR TITLE
Requirements

### PR DESCRIPTION
**Pull request by catherinematus**

### CODE REVIEW

File: requirements.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+// Requirements
+// 1.	The website was created for the possibility of buying books online.
+// 2.	The main page consists of:
+// -	the site logo,
+// -	the link “Shopping Cart” to the shopping cart page,
+// -	the search field with the button “Search”,
+// -	the fields “Email” and “Password” and the button “Login” to log in,
+// -	the link “Sign up” to register,
+// -	presentation of two books (the book cover picture, the author, the book name, the price in $).
```
Comment (VladSelikh) [inline requirements.js:37]: Please do not focus on the design. We assume that design is already known.

File: requirements.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+// 3.	When you click on the site logo, the main page appears/ is updated. Meanwhile the books presented on the main page change.
```
Comment (VladSelikh) [inline requirements.js:38]: Please replace "you" pronoun with the "user" noun in the requirements.

File: requirements.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+// 4.	The search field must be at least 4 characters long.
```
Comment (VladSelikh) [inline requirements.js:39]: Please replace "long" => "in length".

File: requirements.js [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+// 11.	 The field “ZIP code” must be 5 or more digits.
+// 12.	 When you have filled in the field “ZIP code” correctly and clicked the button “Continue”, you go to the user input page. The page consists of:
+// -	the input field First Name,
+// -	the input field Last Name,
+// -	the input field Email,
+// -	the input Password,
+// -	the input field Confirm Password,
+// -	the button “Register”.
+// 13.	 The fields First Name, Email, Password, Confirm Password are mandatory.
+// 14.	 The field Email must have correct email. The name must contain symbols “@” and “.”
```
Comment (VladSelikh) [inline requirements.js:55]: ""Email" field must be of email format" would be enough.

File: requirements.js [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+// -	presentation of two books (the book cover picture, the author, the book name, the price in $).
+// 3.	When you click on the site logo, the main page appears/ is updated. Meanwhile the books presented on the main page change.
+// 4.	The search field must be at least 4 characters long.
+// 5.	The site logo, the link to the cart page and the search field are located at the top of the screen, in the footer. They are present at the all site pages.
+// 6.	When you click the book picture or the book name, you go to the book page. There is a description of the book and the button “Add to Cart” there.
+// 7.	If you are signed up and logged in and when you click the button “Add to Cart”, the book is added to the cart.
+// 8.	You must be signed up and logged in to enter the cart.
+// 9.	When you click the link “Sign up”, you go to the registration page. There is the field “ZIP code” and the button “Continue” there.
+// 10.	 The field “ZIP code” is mandatory.
+// 11.	 The field “ZIP code” must be 5 or more digits.
```
Comment (VladSelikh) [inline requirements.js:46]: What a user will see if he/she tries to submit invalid ZIP code?
Is there any up limit for "ZIP code" field value length?

File: requirements.js (hunk 1 above)
Comment (catherinematus) [reply requirements.js:37]: Ok! I will try))

File: requirements.js (hunk 2 above)
Comment (catherinematus) [reply requirements.js:38]: ok)

File: requirements.js (hunk 3 above)
Comment (catherinematus) [reply requirements.js:39]: ok)

File: requirements.js [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+// c.	 The field “ZIP code” must be 5 digits in length.
+// d.	 When User fills the field “ZIP code” correctly and clicks the button “Continue”, he goes to the User input page. The page consists of:
+// -	the input field "First Name",
+// -	the input field "Last Name",
+// -	the input field "Email",
+// -	the input field "Password",
+// -	the input field "Confirm Password",
+// -	the button “Register”.
+// e.	 The fields "First Name", "Email", "Password", "Confirm Password" are mandatory.
+// f.	 The field "Email" must have correct email.
```
Comment (VladSelikh) [inline requirements.js:29]: I don't see any validation rules for these fields.

File: requirements.js [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+// a.	The main page consists of:
+// -	the site logo,
+// -	the link “Shopping Cart” to the shopping cart page,
+// -	the search field with the button “Search”,
+// -	the fields “Email” and “Password” and the button “Login” to log in,
+// -	the link “Sign up” to register,
+// -	presentation of two books (the book cover picture, the author, the book name, the price in $).
+// b.	When User clicks on the site logo, the main page appears/ is updated. Meanwhile the books presented on the main page change.
+// c.	The search field must be at least 4 characters in length.
+// d.	The site logo, the link to the cart page and the search field are located at the top of the screen, in the footer. They are present at the all site pages.
```
Comment (VladSelikh) [inline requirements.js:13]: Please correct:
`on the all application pages.`

File: requirements.js [hunk 8]
```diff
@@ -0,0 +1,56 @@
+// Requirements
+// The website was created for the possibility of buying books online.
```
Comment (VladSelikh) [inline requirements.js:2]: Please correct:
`... is created`

File: requirements.js [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+// a.   When User clicks the link "Sopping Cart", a page appears with the data of the books that have been added.
+// b.   The data of the added books contains:
+// -    author,
+// -    title,
+// -    quantity (entry field, default value = 1),
+// -    price,
+// -    discount in %,
+// -    discount in $,
+// -    total price in $.
+// c.    When User changes the quantity of the books and clicks the button "Update", the value of the fields "discount %", "discount $", "price $" is updated accoding to the quantity of the books.
```
Comment (VladSelikh) [inline requirements.js:44]: What are the validation rules for a `Quantity` field?

File: requirements.js [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+// 5. Books payment:
+// a.   When User clicks button "Procced to Checkout", he goes to the payment page.
+// b.   Payment page consists of:
+// -    info about the book and total price,
+// -    shipping address,
+// -    payment card details.
+// c.   The payment card section consists of:
+// -    the field "Card Type". By clicking on the icon in this field , User can select his type of card from the drop-down list.
+// -    the field "Card Number". User must fill this field with 16-digits in length number of payment card.
+// -    the fields "Expiration date". User can choose month and year of expited card date from the drop-down list.
```
Comment (VladSelikh) [inline requirements.js:58]: Please correct:
`... of expiration card date`.

File: requirements.js [hunk 11]
```diff
@@ -0,0 +1,60 @@
...
+// -	the input field "First Name",
+// -	the input field "Last Name",
+// -	the input field "Email",
+// -	the input field "Password",
+// -	the input field "Confirm Password",
+// -	the button “Register”.
+// e.	The fields "First Name", "Email", "Password", "Confirm Password" are mandatory.
+// f.   The field "First Name" must be max 255 letters in length, english language, only letters.
+// g.   The field "Password" must be min 4 symbols in length.
+// h.   The field "Confirm Password" must contain the same characters as the field "Password". Input characters must be encoded with dots.
```
Comment (VladSelikh) [inline requirements.js:31]: Please correct:
`"Input characters must be encoded with dots` => `"Input characters must be masked`

**Review by VladSelikh — APPROVED** [review]

**Pull request merged**